code/raw-trie.lisp

```diff
@@ -0,0 +1,148 @@
+(cl:in-package #:spell)
+
+;;; `raw-node' class
+
+(defclass raw-node (node) ())
+
+(defmethod %insert ((object t) (string string) (suffix (eql 0)) (node raw-node))
+  (change-class node 'raw-leaf-node)
+  (%insert object string 0 node))
+
+(defmethod %insert ((object t) (string string) (suffix integer) (node raw-node))
+  (change-class node 'raw-interior-node)
+  (%insert object string suffix node))
+
+;;; `raw-leaf-mixin' class and leaf node protocol methods
+
+(defclass raw-leaf-mixin (leaf-mixin)
+  ((%entries :type   simple-vector
+             :writer (setf %entries))))
+
+(defmethod utilities.print-items:print-items append ((object raw-leaf-mixin))
+  (let ((entries '()))
+    (map-entries (lambda (entry)
+                   (push (class-name (class-of entry)) entries))
+                 object  (%entries object))
+    `(((:entries (:after :children)) " entries: ~{~A~^ ~}" ,entries))))
+
+(defmethod map-entries
+    ((function function) (node raw-leaf-mixin) (entries vector))
+  (declare (type simple-vector entries))
+  (map nil function entries))
+
+(defmethod add-entry ((entry t) (node raw-leaf-mixin) (entries vector))
+  (declare (type simple-vector entries))
+  (let ((new-entries (make-array (+ 1 (length entries)))))
+    (setf (aref new-entries 0)   entry
+          (subseq new-entries 1) entries)
+    new-entries))
+
+#-minimal-raw-trie
```

Comment:
Is this one documented anywhere?

Reply:
No. It would be for the use-case beach asked about. I'm not yet sure whether the option should be kept. What do you think?